exist-core/src/main/java/org/exist/management/impl/JMXAgent.java

```diff
@@ -116,11 +117,7 @@ public synchronized void addMBean(final PerInstanceMBean mbean) throws DatabaseC
         try {
             addMBean(mbean.getName(), mbean);
             if (mbean.getInstanceId() != null) {
-                Deque<ObjectName> stack = registeredMBeans.get(mbean.getInstanceId());
-                if (stack == null) {
-                    stack = new ArrayDeque<>();
-                    registeredMBeans.put(mbean.getInstanceId(), stack);
-                }
+                Deque<ObjectName> stack = registeredMBeans.computeIfAbsent(mbean.getInstanceId(), k -> new ArrayDeque<>());
```

Comment:
I would need some explanation on this in order to be able to review it.

Reply:
> I would need some explanation on this in order to be able to review it.

@line-o  starting with Java 8 the `computeIfAbsent` method as been added, which allows to allow computing an map value based on the given key if not yet exists in one operation using the given lambda expression responsible to calculate the initial value.